test/framework/src/main/java/org/opensearch/test/OpenSearchSingleNodeTestCase.java

```diff
@@ -246,9 +247,16 @@ private Node newNode() {
             .putList(DISCOVERY_SEED_HOSTS_SETTING.getKey()) // empty list disables a port scan for other nodes
             .putList(INITIAL_CLUSTER_MANAGER_NODES_SETTING.getKey(), nodeName)
             .put(FeatureFlags.TELEMETRY_SETTING.getKey(), true)
-            .put(TelemetrySettings.TRACER_ENABLED_SETTING.getKey(), true)
-            .put(nodeSettings()) // allow test cases to provide their own settings or override these
-            .build();
+            .put(TelemetrySettings.TRACER_ENABLED_SETTING.getKey(), true);
+        // allow test cases to provide their own settings or override these
+        settingsBuilder.put(nodeSettings());
+
+        if (Boolean.parseBoolean(settingsBuilder.get(FeatureFlags.CONCURRENT_SEGMENT_SEARCH))
```

Comment:
Not related to this change but it seems like we need to have `featureFlagSettings()` mechanics here as well for OpenSearchSingleNodeTestCase.

Reply:
Yes I also noticed that. Created this [issue](https://github.com/opensearch-project/OpenSearch/issues/8969) for this improvement.